cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/services/http/handlers/ProfileHttpHandlerTest.java

```diff
@@ -355,4 +375,41 @@ public void testNativeProfileImmutable() throws Exception {
     putSystemProfile(ProfileId.NATIVE.getProfile(), Profile.NATIVE, 405);
     deleteSystemProfile(ProfileId.NATIVE.getProfile(), 405);
   }
+
+  @Test
+  public void testPutPermissionChecks() throws Exception {
+    Profile profile = new Profile(PERMISSIONS_TEST_PROFILE, "label", "default permissions testing profile",
+                                         new ProvisionerInfo(MockProvisioner.NAME, Collections.emptyList()));
+    ProfileId profileId = NamespaceId.DEFAULT.profile(PERMISSIONS_TEST_PROFILE);
+    ProfileId systemProfileId = NamespaceId.SYSTEM.profile(PERMISSIONS_TEST_PROFILE);
+    // Verify the UPDATE user does not have permissions to create a profile
+    doAs(UPDATE_PROFILE_USER.getName(), () -> {
+      putProfile(profileId, profile, 403);
```

Comment:
Nit: better to use HttpUrlConnection constants. Much easier for anyone not familiar with codes

Reply:
Acknowledged, I can open a new PR later to fix this since it's a cherry-pick.